drivers/ethernet/eth_w5500.c

```diff
@@ -183,8 +183,17 @@ static int w5500_tx(const struct device *dev, struct net_pkt *pkt)
 	w5500_spi_write_two_bytes(dev, W5500_Sn_TX_WR(0), offset + len);
 
 	w5500_socket_command(dev, 0, W5500_Sn_CR_SEND);
-	if (k_sem_take(&ctx->tx_sem, K_MSEC(10))) {
-		return -EIO;
+
+	k_timepoint_t end = sys_timepoint_calc(K_MSEC(10));
```

Comment:
If not `CONFIG_SPI_ASYNC`, SPI driver will eventually lock it if am not wrong?
but interesting benchmark, thanks. I will test in my end when I find time.